src/OpenSage.Game/Data/Apt/AptFile.cs

```diff
@@ -74,18 +76,40 @@ private void Parse(BinaryReader reader, string parentDirectory)
                     {
                         throw new FileNotFoundException("Cannot find imported file", importPath);
                     }
-                    importApt = AptFile.FromFileSystemEntry(importEntry);
+                    importApt = AptFile.FromFileSystemEntry(importEntry); // If this step is not problematic, initactions should be processed automatically
                     importDict[import.Movie] = importApt;
                 }
 
                 //get the export from that apt and proceed
                 var export = importApt.Movie.Exports.Find(x => x.Name == import.Name);
 
+                // TODO: Unable to import sprites with initactions
                 //place the exported character inside our movie
                 Movie.Characters[(int) import.Character] = importApt.Movie.Characters[(int) export.Character];
             }
+
+            // resolve initactions
+            foreach (var frame in Movie.Frames)
+            {
+                foreach (var item in frame.FrameItems)
+                {
+                    if (item is InitAction ia)
+                    {
+                        var spr = Movie.Characters[(int)ia.Sprite];
+                        if (spr is Sprite sprite)
+                        {
+                            sprite.InitActions = ia.Instructions;
+                        }
+                        else
+                        {
+                            throw new NotImplementedException("ARIENAI!!!!!!!");
```

Comment:
Could you use more descriptive exception messages please? i'm not sure what that means :)

Reply:
@feliwir maybe a bit off topic, but it means _"IMPOSSIBLE!!!"_ [in Japanese](https://en.wiktionary.org/wiki/%E3%81%82%E3%82%8A%E3%81%88%E3%81%AA%E3%81%84#Japanese) ;)